sonar-vbnet-plugin/src/main/java/org/sonar/plugins/vbnet/VbNet.java

```diff
@@ -36,4 +37,13 @@ public VbNet(Configuration configuration) {
     return configuration.getStringArray(VbNetPlugin.FILE_SUFFIXES_KEY);
   }
 
+  @Override
+  public boolean equals(Object o) {
+    return super.equals(o) && configuration == ((VbNet) o).configuration;
```

Comment:
also, to test all conditions, I think you can make a sneaky `TestAbstractLanguage extends AbstractLanguage` that is invoked with `super(VbNetPlugin.LANGUAGE_KEY, VbNetPlugin.LANGUAGE_NAME);` but will fail when casting to a VBNet

at least like this you'll make it clear that it will blow up in case this happens

Reply:
In overall, it's responsibility of the base class unit test to test the base class behavior.

Reply:
Later edit: what I suggested would test the `((VbNet) o).configuration;` , not the base class behavior.

Reply:
it's still a public class and you don't know exactly how it's going to be used by the scanner, right?